tools/engine_tool/test/build_plan_test.dart

```diff
@@ -546,6 +546,102 @@ void main() {
     );
   });
 
+  test('can provide a single extra "--gn-args"', () {
+    final testEnv = TestEnvironment.withTestEngine();
+    addTearDown(testEnv.cleanup);
+
+    final testConfig = TestBuilderConfig();
+    testConfig.addBuild(
+      name: 'linux/host_debug',
+      dimension: TestDroneDimension.linux,
+    );
+
+    final parser = ArgParser();
+    final builds = BuildPlan.configureArgParser(
+      parser,
+      testEnv.environment,
+      configs: {
+        'linux_test_config': testConfig.buildConfig(
+          path: 'ci/builders/linux_test_config.json',
+        ),
+      },
+      help: false,
+    );
+
+    final result = BuildPlan.fromArgResults(
+      parser.parse(['--gn-args', '--foo']),
+      testEnv.environment,
+      builds: builds,
+    );
+    expect(result.extaGnArgs, ['--foo']);
+  });
+
+  test('can provide multiple extra "--gn-arg"s', () {
+    final testEnv = TestEnvironment.withTestEngine();
+    addTearDown(testEnv.cleanup);
+
+    final testConfig = TestBuilderConfig();
+    testConfig.addBuild(
+      name: 'linux/host_debug',
+      dimension: TestDroneDimension.linux,
+    );
+
+    final parser = ArgParser();
+    final builds = BuildPlan.configureArgParser(
+      parser,
+      testEnv.environment,
+      configs: {
+        'linux_test_config': testConfig.buildConfig(
+          path: 'ci/builders/linux_test_config.json',
+        ),
+      },
+      help: false,
+    );
+
+    final result = BuildPlan.fromArgResults(
+      parser.parse(['--gn-args', '--foo', '--gn-args', '--bar']),
```

Comment:
The arg parser doesn't support this being something like `--gn-args --foo --bar --`?

Reply:
It does not.